public/js/rgroups.js

```diff
@@ -0,0 +1,317 @@
+/**
+ * Non-DataTables Index functions.
+ * (The split is there to permit easier switch if we ever yeet datatables from the main UI)
+ */
+const Index = {};
+Index.selectedCategory = "";
+Index.awesomplete = {};
+Index.carouselInitialized = false;
+Index.swiper = {};
+Index.serverVersion = "";
+Index.debugMode = false;
+Index.isProgressLocal = true;
+Index.pageSize = 100;
+Index.carouselMap = new Map();
+Index.baserg;
+
+/**
+ * Initialize the Archive Index.
+ */
+Index.initializeAll = function () {
+    $(document).on("click.open-carousel", ".collapsible-title", Index.toggleCarousel);
+    $(document).on("click.submit-list", "#submit-list", Index.createList);
+    $(document).on("click.add-list", "#add-list", Index.initializeModal);
+    //$(document).on("click.reload-carousel", "#reload-carousel", Index.updateCarousel);
+
+    //Index.updateCarousel();
+
+    // Initialize carousel mode menu
+    $.contextMenu({
+        selector: "#carousel-mode-menu",
+        trigger: "left",
+        build: () => ({
+            callback(key) {
+                console.log(key);
+                var originalElement = $('.context-menu-active');
+                var id = originalElement.parent().parent().parent()[0].id
+                switch (key) {
+                case "edit":
+                    Index.editList(id);
+                    break;
+                case "delete":
+                    Index.removeList(id);
+                    break;
+                }
+            },
+            items: {
+                edit: { name: "Edit", icon: "fas fa-random" },
+                delete: { name: "Delete", icon: "fas fa-envelope-open-text" },
+            },
+        }),
+    });
+
+    Index.initializeGroups();
+    //Index.initializeModal();
+};
+
+Index.initializeModal = function () {
+    // Get the modal
+    var modal = document.getElementById("addl-modal");
+
+    document.getElementById("create-list_form").reset();
+    modal.style.display = "block";
+
+    // When the user clicks anywhere outside of the modal, close it
+    window.onclick = function(event) {
+      if (event.target == modal) {
+        modal.style.display = "none";
+      }
+    }
+}
+
+Index.initializeGroups = function () {
+    //$("#rg_1").remove();
+    Index.baserg = $("#rg_1");
+    let endpoint;
+    endpoint = `/api/reading-group`;
+    Server.callAPI(endpoint, "GET", null, "Error getting carousel data!",
+        (results) => {
+            var list;
+            results.forEach(element => {
+                list = Index.baserg.clone(true);
+                Index.carouselMap.set(element.id, false);
+                list.attr("id", element.id);
+                list.find("#carousel-title").text(element.name);
+                list.appendTo("#list_rg");
+            });
+            $("#rg_1").remove();
+        },
+    );
+}
+
+Index.initializeSortableList = function () {
```

Comment:
Yeah, about the UI, I just want to mention that it's mostly a copy/paste for demo purposes, on my head every client developer would be making the UI whoever they thought would work the best, that's why the API tries to be generic instead of use-case specific. 

Still one aditional comment to add here, the way I thought this part would work(And I didn't add because I didn't want to keep fighting with the UI): First you open the list of groups(Which would be paginated and probably have a searcher), BUT every carrousel would have like 10 or 15 works, and at the end a "Check more" button, which then would take you to a new view, which would not be a carrousel, but probably a more Search like view where the list of works would be paginated, and that way would be easier to explore, here would also be reorder option, which I think will have problems anyway the bigger the number of works in the list, unless a "Move to n position" button is added, so that way if you added the 1000 chapters of OP, but forgot chapter 386, you can just add it at the end -> right click on it -> Move to position 386, and that way you don't have to move it manually either on the pop up nor the carrousel(But because all this is Frontend work, I don't want to opine a lot).

Reply:
Yeah I was just throwing ideas around, fine to focus on the API only if you don't want to fight w/ javascript 👍  

Most reordering UI paradigms fall apart on large groups -- I'll probably have to add some emergency "move to position X" with a textbox as you mentioned.